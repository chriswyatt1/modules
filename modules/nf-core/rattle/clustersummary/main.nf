process RATTLE_CLUSTERSUMMARY {
    tag "$meta.id"
    label 'process_low'

    if (params.enable_conda) {
        error "Conda environments are not set up for rattle (when this module was built). Please use docker or singularity containers."
    }
    container 'ecoflowucl/rattle:v1.0'

    input:
    tuple val(meta), path(reads)
    tuple val(meta), path(clusters)

    output:
    tuple val(meta), path("cluster_summary_${meta.id}.csv"), emit: summary
    path "versions.yml"                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def RATTLE_VERSION = "v1.0" // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    rattle \\
        cluster_summary \\
        $args \\
        -i $reads \\
        -c $clusters \\
        > cluster_summary_${meta.id}.csv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rattle: $RATTLE_VERSION
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def RATTLE_VERSION = "v1.0" // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    touch cluster_summary_${meta.id}.csv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rattle: $RATTLE_VERSION
    END_VERSIONS
    """
}
