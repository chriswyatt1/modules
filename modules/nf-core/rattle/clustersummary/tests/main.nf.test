nextflow_process {

    name "Test Process RATTLE_CLUSTERSUMMARY"
    script "../main.nf"
    process "RATTLE_CLUSTERSUMMARY"

    tag "modules"
    tag "modules_nfcore"
    tag "rattle"
    tag "rattle/clustersummary"
    tag "rattle/cluster"

    test("test-rattle-clustersummary") {

        setup {
            run("RATTLE_CLUSTER") {
                script "../../cluster/main.nf"
                process {
                    """
                    input[0] = [
                        [ id:'test', single_end:false ], // meta map
                        [ file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/nanopore/fastq/test.fastq.gz', checkIfExists: true) ]
                    ]
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/nanopore/fastq/test.fastq.gz', checkIfExists: true) ]
				]
                input[1] = RATTLE_CLUSTER.out.clusters
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test-rattle-clustersummary-stub") {
        options '-stub'

        setup {
            run("RATTLE_CLUSTER") {
                script "../../cluster/main.nf"
                process {
                    """
                    input[0] = [
                        [ id:'test', single_end:false ], // meta map
                        [ file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/nanopore/fastq/test.fastq.gz', checkIfExists: true) ]
                    ]
                    """
                }
            }
        }
        
        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/nanopore/fastq/test.fastq.gz', checkIfExists: true) ]
                ]
                input[1] = RATTLE_CLUSTER.out.clusters
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
