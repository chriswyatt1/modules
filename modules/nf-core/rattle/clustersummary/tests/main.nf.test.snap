{
    "test-rattle-clustersummary": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "cluster_summary_test.csv:md5,45760f8b70b8e0ad0b53d8a617bb0a32"
                    ]
                ],
                "1": [
                    "versions.yml:md5,49706d301fbc166dc1e61b6c665d1f35"
                ],
                "summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "cluster_summary_test.csv:md5,45760f8b70b8e0ad0b53d8a617bb0a32"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,49706d301fbc166dc1e61b6c665d1f35"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-11T20:04:17.059056"
    },
    "test-rattle-clustersummary-stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "cluster_summary_test.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,49706d301fbc166dc1e61b6c665d1f35"
                ],
                "summary": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "cluster_summary_test.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,49706d301fbc166dc1e61b6c665d1f35"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-11T20:04:24.600674"
    }
}